shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPersistService.java

```diff
@@ -0,0 +1,175 @@
+        return Optional.ofNullable(schedulerMap.get(shardingItem));
+    }
+    
+    /**
+     * List job schedules.
+     *
+     * @param jobId job id
+     * @return job schedules belong this job id
+     */
+    public static Collection<RuleAlteredJobScheduler> listJobSchedule(final String jobId) {
+        Map<Integer, RuleAlteredJobScheduler> schedulerMap = JOB_SCHEDULER_MAP.computeIfAbsent(jobId, key -> new ConcurrentHashMap<>());
+        return schedulerMap.values();
+    }
+    
+    /**
+     * Remove job schedule map by job id.
+     *
+     * @param jobId job id
+     */
+    public static void removeJobSchedule(final String jobId) {
+        log.info("Remove job schedule by job id: {}", jobId);
+        JOB_SCHEDULER_MAP.remove(jobId);
+        JOB_PERSIST_MAP.remove(jobId);
+    }
+    
+    /**
+     * Add job schedule.
+     *
+     * @param jobId job id
+     * @param shardingItem sharding item
+     * @param jobScheduler job schedule
+     */
+    public static void addJobSchedule(final String jobId, final int shardingItem, final RuleAlteredJobScheduler jobScheduler) {
+        log.info("Add job schedule, jobId={}, shardingItem={}", jobId, shardingItem);
+        JOB_SCHEDULER_MAP.computeIfAbsent(jobId, key -> new ConcurrentHashMap<>()).put(shardingItem, jobScheduler);
+        JOB_PERSIST_MAP.computeIfAbsent(jobId, key -> new ConcurrentHashMap<>()).put(shardingItem, new JobPersistIntervalParameter(jobId, shardingItem));
+    }
+    
+    /**
+     * Persist job process, may not be implemented immediately, depending on persist interval.
+     *
+     * @param jobId job id
+     * @param shardingItem sharding item
+     * @param currentTimeMills current time mills
+     */
+    public static void triggerPersist(final String jobId, final int shardingItem, final long currentTimeMills) {
+        Map<Integer, JobPersistIntervalParameter> intervalParamMap = JOB_PERSIST_MAP.getOrDefault(jobId, Collections.emptyMap());
+        JobPersistIntervalParameter parameter = intervalParamMap.get(shardingItem);
+        if (null == parameter) {
+            log.debug("Persist interval parameter is null, jobId={}, shardingItem={}", jobId, shardingItem);
+            return;
+        }
+        parameter.getAlreadyPersisted().compareAndSet(true, false);
+        if ((currentTimeMills - parameter.getPersistTime().get()) < MIN_PERSIST_INTERVAL_MILLIS) {
+            return;
+        }
+        if (parameter.getLock().tryLock()) {
+            try {
+                SINGLE_EXECUTOR.execute(() -> persist(jobId, shardingItem, currentTimeMills, parameter));
+                // CHECKSTYLE:OFF
+            } catch (final RuntimeException ex) {
+                // CHECKSTYLE:ON
+                log.error("Persist job process failed, job id: {}, sharding item: {}", jobId, shardingItem, ex);
+            } finally {
+                parameter.getLock().unlock();
+            }
+        }
```

Comment:
1, Is it necessary to do persist here? Since `JOB_PERSIST_EXECUTOR` exists.

2, There might be duplicated persistenance on high concurrent invocations.